### PR TITLE
Feature/gallery view

### LINKED ISSUE #16: As a user, I should be able to upload gallery photos from device storage.

### LINKED ISSUE #31: As a user, I should see an Obituary secondary title on the Obituary page.

### LINKED ISSUE #41: As a user, I should see a gallery page of the photos I have uploaded.

### LINKED ISSUE #42: As a user, I should see a secondary header on the gallery page that says "Gallery".

### PR DESCRIPTION
**Pull request by AshleyOh-bit**
## Description
- Built gallery view as a stained glass window
- hooked up gallery view to add a photo button
- created dynamic routes for add photos depending on profile or gallery photo type
- re-styling obit page and added obituary title
- passed mock data to gallery view from app to simulate data flow post fetch

## Concerns

-  Any bugs still present?
- Not to my knowledge
-  Next steps?
- Gallery needs to be hooked up to fetch call for photos
- Photo add for gallery needs to be hooked up to post endpoint

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Refactor (styling)

## Checklist:
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

## Issues closed:
- closes #31 
- closes #16 
- closes #41 
- closes #42

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -66,7 +70,14 @@ const App = () => {
           <Route exact path="/countdown" render={() => <CountdownPane etd={user.attributes.etd} err={error} dob={user.attributes.date_of_birth}/>}/>
           <Route exact path="/obituary" render={() => <ObitPane obit={user.attributes.obituary}/>}/>
           <Route exact path="/executors" render={() => <ExecutorPane executor={user.attributes.executor}/>}/>
-          <Route exact path="/photoadd" render={() => <PhotoAdd updateProfilePicture={updateProfilePicture} currentProfilePic={user.attributes.profile_picture_url}/>}/>
+          <Route exact path="/add-photo/profile" render={() => <PhotoAdd updateProfilePicture={updateProfilePicture} currentProfilePic={user.attributes.profile_picture_url} type={"profile"}/>}/>
```
Comment (NoahZinter) [inline src/components/App/App.js:73]: These paths refer to paths you're defining within the FE app? To redirect to a given component?

File: src/components/App/App.js [hunk 2]
```diff
@@ -66,7 +70,14 @@ const App = () => {
...
+          <Route exact path="/add-photo/gallery" render={() => <PhotoAdd addGalleryPhoto={addGalleryPhoto} profPhoto={user.attributes.profile_picture_url} type={"gallery"}/>}/> 
+          <Route exact path="/gallery" render={() => <GalleryPane profPhoto={user.attributes.profile_picture_url} galPhotos={galleryPhotos}/>}/>
+          <Route exact path="/add-photo/:type" render={({ match }) => 
+            { const whichType = match.params.type === "profile" ? <PhotoAdd updateProfilePicture={updateProfilePicture} currentProfilePic={user.attributes.profile_picture_url} type={"profile"}/>
+                                                                : <PhotoAdd addGalleryPhoto={addGalleryPhoto} profPhoto={user.attributes.profile_picture_url} type={"gallery"}/>
+            }
```
Comment (NoahZinter) [inline src/components/App/App.js:79]: Where would I look for the fetch call associated with this?

File: src/components/App/App.js (hunk 1 above)
Comment (AshleyOh-bit) [reply src/components/App/App.js:73]: yep! This is how we simulate the kind of navigation a user is familiar with i.e. with the URL changing. It allows them to send the URLs to friends etc too!

File: src/components/App/App.js [hunk 3]
```diff
@@ -12,27 +12,31 @@ import galleryImg from "../../assets/gallery.png"
...
 import { ExecutorPane } from "../ExecutorPane/ExecutorPane";
 import PhotoAdd from "../PhotoAdd/PhotoAdd";
+import { GalleryPane } from "../GalleryPane/GalleryPane";
 import ObitPane from "../ObitPane/ObitPane";
 import { Switch, NavLink, Link, Route } from 'react-router-dom';
 
 const App = () => {
   const [user, setUser] = useState({})
   const [error, setError] = useState("")
+  const [galleryPhotos, setGalleryPhotos] = useState(["http://c.files.bbci.co.uk/11382/production/_119803507_mediaitem119803506.jpg", "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQAmmgkZCiMVGYv9i0A82kdTe-I5JogeRNzog&usqp=CAU", "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQz3xXj4Q8qHmazIiBUgmeB-qYugUaZrvN7Mw&usqp=CAU"])
```
Comment (NoahZinter) [inline src/components/App/App.js:22]: Does useState, used in this way, allow you to set mock data?

File: cypress/integration/change_profile_picture.js [hunk 4]
```diff
@@ -36,6 +36,8 @@ describe("Change Profile Picture", () => {
             })
             .get(".photo-edit-button")
             .click()
+            cy.get(".sun > img")
```
Comment (tylrs) [inline cypress/integration/change_profile_picture.js:39]: Why was this added to the test?

File: src/components/App/App.js (hunk 2 above)
Comment (AshleyOh-bit) [reply src/components/App/App.js:79]: Taylor will be implementing that fetch call on his branch in the next hour or so. It should be happening inside of the getUser() function.

File: src/components/App/App.js (hunk 3 above)
Comment (AshleyOh-bit) [reply src/components/App/App.js:22]: yep! It also allows us to set state in a functional component, which makes our app a little "lighter".

File: src/components/ObitPane/ObitPane.js [hunk 5]
```diff
@@ -27,43 +27,46 @@ const ObitPane = ({obit}) => {
...
-                        value={obituary}
-                        onChange={(e) => {handleChange(e)}}
-                        maxLength='500'
-                    >
-                    </textarea>   
-                    
-                : <article className='obit-text'>{obituary}</article>
-            }
-        </section>
+        <>
```
Comment (NoahZinter) [inline src/components/ObitPane/ObitPane.js:30]: What do these empty carrots do?

File: cypress/integration/change_profile_picture.js (hunk 4 above)
Comment (AshleyOh-bit) [reply cypress/integration/change_profile_picture.js:39]: It was added to the test because the button now redirects to the dashboard. This is because of the conditional rendering needed inside the component so it can be used by Gallery too!

File: cypress/integration/gallery_spec.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+              fileContent: fileContent,
+              fileName: "william.png",
+              mimeType: "image/png"
+          })
+          .get(".photo-edit-button")
+          .click()
+        cy.get(".g-14")
+          .children("img")
+          .should("be.visible")
+          .should("have.attr", "src").should("include", "blob:http://localhost:3000")
```
Comment (tylrs) [inline cypress/integration/gallery_spec.js:88]: So the only thing with this test is that I'm just making sure that the first part of the source is correct because the rest is changing. You would not be able to test if you had uploaded different photos. just FYI

File: src/components/ObitPane/ObitPane.js (hunk 5 above)
Comment (AshleyOh-bit) [reply src/components/ObitPane/ObitPane.js:30]: This is called a fragment - basically this is a JSX element that doesn't appear on the page, but it still allows the elements to be grouped together and returned as a unit to the virtual DOM.

File: src/components/App/App.js (hunk 1 above)
Comment (tylrs) [reply src/components/App/App.js:73]: Sorry did not realize the submit button was taking the user to the dashboard

**Review by NoahZinter — APPROVED** [review]

File: src/components/App/App.js [hunk 7]
```diff
@@ -66,7 +70,14 @@ const App = () => {
           <Route exact path="/countdown" render={() => <CountdownPane etd={user.attributes.etd} err={error} dob={user.attributes.date_of_birth}/>}/>
           <Route exact path="/obituary" render={() => <ObitPane obit={user.attributes.obituary}/>}/>
           <Route exact path="/executors" render={() => <ExecutorPane executor={user.attributes.executor}/>}/>
-          <Route exact path="/photoadd" render={() => <PhotoAdd updateProfilePicture={updateProfilePicture} currentProfilePic={user.attributes.profile_picture_url}/>}/>
+          <Route exact path="/add-photo/profile" render={() => <PhotoAdd updateProfilePicture={updateProfilePicture} currentProfilePic={user.attributes.profile_picture_url} type={"profile"}/>}/>
+          <Route exact path="/add-photo/gallery" render={() => <PhotoAdd addGalleryPhoto={addGalleryPhoto} profPhoto={user.attributes.profile_picture_url} type={"gallery"}/>}/>
```
Comment (tylrs) [inline src/components/App/App.js:74]: I'm assuming you changed this profPhoto in the PhotoAdd component as well

File: src/components/App/App.js [hunk 8]
```diff
@@ -66,7 +70,14 @@ const App = () => {
...
           <Route exact path="/obituary" render={() => <ObitPane obit={user.attributes.obituary}/>}/>
           <Route exact path="/executors" render={() => <ExecutorPane executor={user.attributes.executor}/>}/>
-          <Route exact path="/photoadd" render={() => <PhotoAdd updateProfilePicture={updateProfilePicture} currentProfilePic={user.attributes.profile_picture_url}/>}/>
+          <Route exact path="/add-photo/profile" render={() => <PhotoAdd updateProfilePicture={updateProfilePicture} currentProfilePic={user.attributes.profile_picture_url} type={"profile"}/>}/>
+          <Route exact path="/add-photo/gallery" render={() => <PhotoAdd addGalleryPhoto={addGalleryPhoto} profPhoto={user.attributes.profile_picture_url} type={"gallery"}/>}/> 
+          <Route exact path="/gallery" render={() => <GalleryPane profPhoto={user.attributes.profile_picture_url} galPhotos={galleryPhotos}/>}/>
+          <Route exact path="/add-photo/:type" render={({ match }) => 
+            { const whichType = match.params.type === "profile" ? <PhotoAdd updateProfilePicture={updateProfilePicture} currentProfilePic={user.attributes.profile_picture_url} type={"profile"}/>
```
Comment (tylrs) [inline src/components/App/App.js:77]: Nice use of match.params!

File: src/components/App/App.js [hunk 9]
```diff
@@ -66,7 +70,14 @@ const App = () => {
           <Route exact path="/countdown" render={() => <CountdownPane etd={user.attributes.etd} err={error} dob={user.attributes.date_of_birth}/>}/>
           <Route exact path="/obituary" render={() => <ObitPane obit={user.attributes.obituary}/>}/>
           <Route exact path="/executors" render={() => <ExecutorPane executor={user.attributes.executor}/>}/>
-          <Route exact path="/photoadd" render={() => <PhotoAdd updateProfilePicture={updateProfilePicture} currentProfilePic={user.attributes.profile_picture_url}/>}/>
+          <Route exact path="/add-photo/profile" render={() => <PhotoAdd updateProfilePicture={updateProfilePicture} currentProfilePic={user.attributes.profile_picture_url} type={"profile"}/>}/>
+          <Route exact path="/add-photo/gallery" render={() => <PhotoAdd addGalleryPhoto={addGalleryPhoto} profPhoto={user.attributes.profile_picture_url} type={"gallery"}/>}/> 
+          <Route exact path="/gallery" render={() => <GalleryPane profPhoto={user.attributes.profile_picture_url} galPhotos={galleryPhotos}/>}/>
+          <Route exact path="/add-photo/:type" render={({ match }) =>
```
Comment (tylrs) [inline src/components/App/App.js:76]: Why is there a dynamic route and two static routes for add-photo?

File: src/components/Header/Header.js [hunk 10]
```diff
@@ -5,7 +5,7 @@ import userPicture from '../../assets/user.png';
 
 const Header = ({profilePic}) => {
     let pic;
-    profilePic.length ? pic = profilePic : pic = userPicture;
+    profilePic ? pic = profilePic : pic = userPicture;
```
Comment (tylrs) [inline src/components/Header/Header.js:8]: I just changed this on my end too!

File: src/components/App/App.js (hunk 9 above)
Comment (tylrs) [reply src/components/App/App.js:76]: We can fix the dynamic route on a future branch

**Review by tylrs — APPROVED** [review]
